raster-foundry/raster-foundry
# Make org de/activation work. Act on membership request/invitations.

## Overview

This PR makes frontend and backend changes so that:
 * users will know their roles (i.e. ADMIN or MEMBER) when listing organizations and teams they belong to;
 * for admins, show invitation, request status on user list page on organization level and on team level, and enable these buttons' actions;
 * for admins, make organization de/activation work;
 * sort users in organization and team lists by membership status and by group role so that pending approval and pending invitation items show on top and that admin items show before members;
 * Fix user profile picture broken bug on profile pages.

~Marked as WIP since acting on membership invitation/request on team level has not been implemented yet.~

### Checklist

- ~Styleguide updated, if necessary~
- ~[Swagger specification](https://github.com/raster-foundry/raster-foundry-api-spec) updated~
- ~Symlinks from new migrations present or corrected for any new migrations~
- ~Any content changes are properly templated using `BUILDCONFIG.APP_NAME`~
- ~Any new SQL strings have tests~

### Demo

<img width="1680" alt="screen shot 2018-06-27 at 11 30 52 am" src="https://user-images.githubusercontent.com/16109558/41984021-969315d2-79fd-11e8-96bf-5a2aa4700c3b.png">


## Testing Instructions

 * Log in as an admin of a platform. Go to organization list page under platform management, check if organization activation/deactivation works
 * Check if user profile picture shows up on user profile pages
 * Log in as an admin of an organization. In DB, modify some organization users' membership status so that they are either `REQUESTED` or `INVITED`. Go to organization user list page, check if these pending statuses show up on top, then approved admins, and then approved members. Click on those approval/denial/cancelation buttons, and see if they work as you expect.
 * Log in as an admin of a team. Do similar tests as the previous case.
 * With any role, get to the user list page of the team/organization you belong to, make sure you see your role showing up in the list as well.

Closes #3593 
Closes #3594 
Closes #3595
